src/js/tracker.js

```diff
@@ -796,6 +845,215 @@
 		}
 
 		/**
+		 * Creates a context from the window['optimizely'].data.experiments object
+		 *
+		 * @return array Experiment contexts
+		 */
+		function getOptimizelyExperimentContexts() {
+			var experiments = window['optimizely'].data.experiments;
+			if (experiments) {
+				var contexts = [];
+
+				for (var key in experiments) {
```

Comment:
For loops like this are a good example of when we actually should use hasOwnProperty. If somebody else has executed something like [Object.prototype.watch](https://gist.github.com/eligrey/384583) on the page, the new "watch" property will be included in this loop. Presumably `Object.prototype.watch.code` evaluates to `undefined`, so we would end up with an extra context with no populated fields.